### PR TITLE
Bug/6593/fix send email for user in patch update status

### PR DESCRIPTION
**Pull request by VasylyshynDmytro**
[Link on task from board](https://github.com/orgs/ita-social-projects/projects/43/views/6?filterQuery=API&pane=issue&itemId=69330459&issue=ita-social-projects%7CGreenCity%7C6593)

## Summary Of Changes :fire:
added the logic of sending letters to the user if the status of the location is changed

image

## Added
* _&lt;detail item of what was added&gt;_

- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

**Review by Maryna-511750 — APPROVED** [review]

**Review by KizerovDmitriy — APPROVED** [review]

File: service/src/test/java/greencity/service/EmailServiceImplTest.java [hunk 1]
```diff
@@ -29,13 +32,14 @@
...
 import java.util.*;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.Executors;
 
 import static greencity.ModelUtils.getSubscriberDto;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/EmailServiceImplTest.java]: please do nor use `*`

File: core/src/main/resources/messages_uk.properties [hunk 2]
```diff
@@ -54,4 +54,5 @@ advice.for.block=\u0412\u0438\u0020\u043E\u0442\u0440\u0438\u043C\u0430\u043B\u0
...
-read.more=\u0427\u0418\u0422\u0410\u0422\u0418 \u0414\u0410\u041b\u0406
\ No newline at end of file
+read.more=\u0427\u0418\u0422\u0410\u0422\u0418 \u0414\u0410\u041b\u0406
+your.place.status.changed=\u0421\u0442\u0430\u0442\u0443\u0441\u0020\u0432\u0430\u0448\u043E\u0433\u043E\u0020\u043C\u0456\u0441\u0446\u044F\u0020\u0431\u0443\u0432\u0020\u043E\u043D\u043E\u0432\u043B\u0435\u043D\u0438\u0439\u002E\u0020\u0412\u0438\u0020\u043E\u0442\u0440\u0438\u043C\u0430\u043B\u0438\u0020\u0446\u0435\u0439\u0020\u043B\u0438\u0441\u0442\u002C\u0020\u0442\u043E\u043C\u0443\u0020\u0449\u043E\u0020\u0441\u0442\u0430\u0442\u0443\u0441\u0020\u0432\u0430\u0448\u043E\u0433\u043E\u0020\u043C\u0456\u0441\u0446\u044F\u0020\u0431\u0443\u043B\u043E\u0020\u0437\u043C\u0456\u043D\u0435\u043D\u043E\u002E\u0020\u0414\u044F\u043A\u0443\u0454\u043C\u043E\u002C\u0020\u0449\u043E\u0020\u043A\u043E\u0440\u0438\u0441\u0442\u0443\u0454\u0442\u0435\u0441\u044C\u0020\u043D\u0430\u0448\u0438\u043C\u0020\u0441\u0435\u0440\u0432\u0456\u0441\u043E\u043C\u0021
```
Comment (ChernenkoVitaliy) [inline core/src/main/resources/messages_uk.properties]: You added this UA constant message but in the EmailServiceImpl in method _sendPlaceStatusChangeNotification_ you set  `model.put(EmailConstants.LANGUAGE, "en");` It seems like this constant will be never used.

File: service/src/main/java/greencity/service/EmailServiceImpl.java [hunk 3]
```diff
@@ -317,6 +318,19 @@ private Map<String, Object> buildModelMapForPasswordRestore(Long userId, String
         return model;
     }
 
+    @Override
+    public void sendPlaceStatusChangeNotification(PlaceStatusChangeDto dto) {
```
Comment (ChernenkoVitaliy) [inline service/src/main/java/greencity/service/EmailServiceImpl.java:329]: I'm not sure but, maybe it makes sense before sending an email too check if the user exists in DB.
Do something like this
```
if (userRepo.existsUserByEmail(dto.getUserEmail())) {
     throw new WrongEmailException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + dto.getUserEmail());
}
```
Cause I've tried to pass an email address of not a registered user and the email was sent successfully.

**Review by ChernenkoVitaliy — APPROVED** [review]

File: service/src/main/java/greencity/service/EmailServiceImpl.java [hunk 4]
```diff
@@ -317,6 +325,21 @@ private Map<String, Object> buildModelMapForPasswordRestore(Long userId, String
...
+        User user = userRepo.findByEmail(dto.getEmail())
+            .orElseThrow(() -> new RuntimeException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + dto.getEmail()));
+        model.put(EmailConstants.CLIENT_LINK, clientLink);
+        model.put(EmailConstants.USER_NAME, dto.getUserName());
+        model.put(EmailConstants.PLACE_NAME, dto.getPlaceName());
+        model.put(EmailConstants.PLACE_STATUS, dto.getNewStatus().name());
+        model.put(EmailConstants.LANGUAGE, user.getLanguage().getCode());
+
+        String template = createEmailTemplate(model, EmailConstants.PLACE_STATUS_CHANGE_PAGE);
+        sendEmail(dto.getEmail(), "Status of your place has been updated", template);
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EmailServiceImpl.java]: Please move `"Status of your place has been updated"` to a constant

**Review by RomaMocherniuk — APPROVED** [review]

**Pull request merged**